src/filter.c

```diff
@@ -12,21 +12,32 @@
 #include <assert.h>
 #include <zlib.h>
 #include <getopt.h>
+#include <time.h>
 
 #define block_size 2048
 #define unsafe_block_size 4096
 
 int threshold;
+char *r1i_path = NULL, *r2i_path = NULL, *r1o_path = NULL, *r2o_path = NULL, *stats_file = NULL;
+int read_pairs_checked = 0, read_pairs_removed = 0, read_pairs_remaining = 0;
 
-static void _log(char* msg) {
-    printf("[fastq_filterer] %s\n", msg);
+
+static void timestamp() {
+
+    time_t t = time(NULL);
+    struct tm* now = localtime(&t);
+    
+    printf(
+        "[%i-%i-%i %i:%i:%i][fastq_filterer] ",
+        now->tm_year + 1900, now->tm_mon + 1, now->tm_mday, now->tm_hour, now->tm_min,  now->tm_sec
+    );
```

Comment:
This was because `_log` could only log a single hardcoded string. To output a dynamically-generated message, we previously had to build up a formatted string with `sprintf`, which was a bit unwieldy when we just wanted to log the value of an int or a file path. Instead, we can now just use `printf`:
```c
int x = 3;
char* file_path = "path/to/a/file";
timestamp();
printf("x is %i, and file_path is %s\n", x, file_path);
```
Ideally we would wrap printf, but we'd need to dynamically pass `...` varargs to printf, which I don't think is possible in C.